### PR TITLE
Merge tom-dev

### PR DESCRIPTION
**Pull request by tomrittervg**
I rebased this on master (took me a bit).  `./test.py` passes.  

Something is surely wrong in this big set of changes though - please review!

### CODE REVIEW

**Comment by tomrittervg** [comment]
Also, there's some todos in this set of changes I didn't open tickets for, we need to do that before we merge.

**Comment by metalcanine** [comment]
I'm on it!

**Comment by tomrittervg** [comment]
I added a commit for the lint errors; but it doesn't seem to have retrigger the lint job...

**Comment by metalcanine** [comment]
It looks like they were rerun it's just finding these now:
```log
./components/hg.py:16:13: E128 continuation line under-indented for visual indent
./components/db.py:25:1: E302 expected 2 blank lines, found 1
./components/db.py:43:1: E305 expected 2 blank lines after class or function definition, found 1
./components/bugzilla.py:25:13: E128 continuation line under-indented for visual indent
./components/bugzilla.py:37:1: W391 blank line at end of file
./tests/bugzilla.py:9:1: E402 module level import not at top of file
./tests/bugzilla.py:11:1: E402 module level import not at top of file
./tests/bugzilla.py:12:1: E402 module level import not at top of file
./tests/bugzilla.py:13:1: E402 module level import not at top of file
./tests/bugzilla.py:15:1: E402 module level import not at top of file
./tests/bugzilla.py:16:1: E402 module level import not at top of file
./tests/bugzilla.py:17:1: E402 module level import not at top of file
./tests/bugzilla.py:61:1: E303 too many blank lines (3)
./tests/automation_configuration.py:9:1: E402 module level import not at top of file
./tests/automation_configuration.py:11:1: E402 module level import not at top of file
./tests/automation_configuration.py:74:9: F841 local variable 'u' is assigned to but never used
```

Also I checked the tests locally and the pass for me too! They won't pass in CI until we add a mysql database in CI for it to interact with

**Comment by metalcanine** [comment]
Can you also run `poetry add cryptography` and add that in a commit?

**Comment by tomrittervg** [comment]
I fixed some of those new lint errors; but the module import ones are happening because of the `sys.path.append` - that's needed where it is to make the __main__ work if you run the singular test directly.  I'm not sure what the solution there is.  Not support that?

**Comment by tomrittervg** [comment]
I ran `poetry add cryptography` and have no idea what it did but here is the commit.

**Comment by metalcanine** [comment]
It looks like the only remaining ones are the E402 errors. If you want we can add that to the list of ignored lint errors (currently just E501)

Thanks for adding the cryptography dependency!

**Comment by metalcanine** [comment]
To disable E402 just add `,E402` after `E501` here https://github.com/mozilla-services/updatebot/blob/95a3ec1f2873ed8e55d66cfc82f0babaaa629b5e/.circleci/config.yml#L46

(sorry this is turning out to be such a pain)

File: .circleci/config.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
           command: poetry check
           name: dependencies
       - run:
-          command: poetry run flake8 --ignore=E501 .
+          command: poetry run flake8 --ignore=E501 --ignore=E402 .
```
[the remark this answers is not in the document]
Comment (tomrittervg) [reply .circleci/config.yml]: fixed

**Pull request merged**